dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -291,8 +291,21 @@ public long remaining() {
 
   @Override
   public int positionedRead(long pos, byte[] b, int off, int len) throws IOException {
-    return readInternal(new ByteArrayTargetBuffer(b, off), off, len, ReadType.READ_INTO_BYTE_ARRAY,
-        pos, true);
+    try {
```

Comment:
If there is a permission deny thrown, it should not fall back to ufs read. Is there a way to handle that?

And there is also a config to disable the fall-back read, can we check here?

Reply:
why there is a permission deny? the caching file is on local disk, this input stream is just for presto local cache

Reply:
In my impression the read should read local disk first, then worker, then fallback to UFS? 
Wouldn't the local file be the highest priority?

mExternalFileInStreamOpener is trying to open a stream to UFS, am I right?

Reply:
this class is used by presto's local cache only, there is no integration with dora worker so far.   mExternalFileInStreamOpener is taking the input stream from presto.

Reply:
Understood.